### PR TITLE
Seed handling for Sampling

### PR DESCRIPTION
**Pull request by jduerholt**
This PR adds random seed handling for sampling.

### CODE REVIEW

File: bofire/strategies/strategy.py [hunk 1]
```diff
@@ -26,6 +26,15 @@ def __init__(
...
         self._candidates = None
 
+    def _get_seed(self) -> int:
+        """Returns an integer sampled from the strategies random number generator,
+        that can be used to seed dependent generators.
+
+        Returns:
+            int: random seed.
+        """
+        return int(self.rng.integers(1, 100000))
```
Comment (dlinzner-bcs) [inline bofire/strategies/strategy.py:36]: Out of interest: This is just a very large number?

Comment (dlinzner-bcs) [reply bofire/strategies/strategy.py:36]: can you use `sys.maxsize`? https://stackoverflow.com/questions/7604966/maximum-and-minimum-values-for-ints

**Review by dlinzner-bcs — APPROVED** [review]
Looks good to me. Only the "large numbers" are arbitrary, so the space seeds for generating random generators will be artifically shrinked. I can however think of no consequence of this :D

File: bofire/strategies/strategy.py (hunk 1 above)
Comment (jduerholt) [reply bofire/strategies/strategy.py:36]: Yep, it is just a large number, botorch is using the same I think, for this reason, I would stay with this one.

**Pull request merged**